doc/releases/changelog-0.9.0.md

```diff
@@ -210,32 +224,38 @@
   catalyst-cli --tool=llc llvm-ir.ll -o output.o
   ```
```

Comment:
@mehrdad2m I noticed that there is no documentation for `catalyst-cli` aside from the changelog entry, which will likely make it hard for developers to use/us to maintain it.

Likely the best approach here is to add a catalyst-cli page to the `API` section of the docs:

<img width="262" alt="image" src="https://github.com/user-attachments/assets/b07621ed-bbf9-4d75-904c-ae714aafa17c">

with the usage/documentation of the CLI tool pulled _directly_ from the source code/`--help` output, to make it more maintainable (does `catalyst-cli --help` work?)

Reply:
@mehrdad2m is away this week, but I can have a look and add the usage instructions.

Reply:
thanks Joey :)